lib/action_view/component/base.rb

```diff
@@ -219,7 +219,13 @@ def template_errors
           @template_errors ||=
             begin
               errors = []
-              errors << "#{self} must implement #initialize." if source_location.nil?
+
+              unless const_source_location_supported?
+                # To support Ruby < 2.7.0, require `#initialize` to be defined so that we can use
+                # `method#source_location`.
+                errors << "#{self} must implement #initialize." if source_location.nil?
+              end
+
               errors << "Could not find a template file for #{self}." if templates.empty?
```

Comment:
note that `#templates` should be empty if `#source_location` is `nil` (see above, [base.rb#L201](https://github.com/github/actionview-component/pull/174/files#diff-4ea3d395eff742fa0d8952d23a16d377L201)), so if we're using 2.7+ and `#source_location` is nil for some other reason, this is the error that should be raised. 

I'm not sure what other reasons `source_location` could be nil, so I'm not currently testing this.